File: Testing/Code for Testing/Early Basic Syntax Testing/inherit/test_inherit.py
from dataclasses import dataclass, field

def _ul_sort(arr, n): arr[:n] = sorted(arr[:n])

class _ul_Exception(Exception):
    def __init__(self, type_name, message):
        self.type_name = type_name
        self.message = message
        super().__init__(message)

@dataclass
class Animal:
    name: str = ""
    age: int = 0

@dataclass
class Dog(Animal):
    breed: str = ""

@dataclass
class Puppy(Dog):
    weight: int = 0


def describe(d):
    print(d.name)
    print(d.age)
    print(d.breed)

def main():
    a = Animal()
    a.name = 'Cat'
    a.age = 5
    print(a.name)
    print(a.age)
    d = Dog()
    d.name = 'Rex'
    d.age = 3
    d.breed = 'Labrador'
    print(d.name)
    print(d.age)
    print(d.breed)
    describe(d)
    p = Puppy()
    p.name = 'Tiny'
    p.age = 1
    p.breed = 'Beagle'
    p.weight = 8
    print(p.name)
    print(p.age)
    print(p.breed)
    print(p.weight)
    return 0


if __name__ == "__main__":
    main()

File: Testing/Code for Testing/Early Basic Syntax Testing/inherit/test_inherit.js
function createAnimal() {
    return { name: "", age: 0 };
}

function createDog() {
    return { ...createAnimal(), breed: "" };
}

function createPuppy() {
    return { ...createDog(), weight: 0 };
}


function describe(d) {
    console.log(d.name);
    console.log(d.age);
    console.log(d.breed);
}

function main() {
    let a = createAnimal();
    a.name = "Cat";
    a.age = 5;
    console.log(a.name);
    console.log(a.age);
    let d = createDog();
    d.name = "Rex";
    d.age = 3;
    d.breed = "Labrador";
    console.log(d.name);
    console.log(d.age);
    console.log(d.breed);
    describe(d);
    let p = createPuppy();
    p.name = "Tiny";
    p.age = 1;
    p.breed = "Beagle";
    p.weight = 8;
    console.log(p.name);
    console.log(p.age);
    console.log(p.breed);
    console.log(p.weight);
    return 0;
}

function __ul_sort(arr, n) { const sub = arr.slice(0, n).sort((a, b) => a < b ? -1 : a > b ? 1 : 0); for (let i = 0; i < n; i++) arr[i] = sub[i]; }

main();
